src/sql/src/session/hint.rs

```diff
@@ -45,10 +59,24 @@ pub enum ApplicationNameHint {
     Dbt(Private),
     /// Request came from our web console.
     WebConsole(Private),
+    /// Request came from the SQL shell in our web console.
+    WebConsoleShell(Private),
     /// Request came from the `psql` shell spawned by `mz`.
     MzPsql(Private),
     /// Request came from our Fivetran Destination,
     MaterializeFivetranDestination(Private),
+    /// Request came from a version of our Terraform provider.
+    TerraformProviderMaterialize(Private),
+    /// Request came from TablePlus.
+    TablePlus(Private),
+    /// Request came from a version of DataGrip.
+    DataGrip(Private),
+    /// Request came from a version of DBeaver.
+    DBeaver(Private),
+    /// Request came from our Visual Studio Code integration.
+    MzVscode(Private),
+    /// Request came from our Grafana integration.
+    MzGrafanaIntegration(Private),
```

Comment:
Do we need to be worried about what the comment above says?

> /// Note: [`ApplicationNameHint`] gets logged as a label in our Prometheus metrics, and for
> /// labels we need to be conscious of the cardinality, so please be careful with how many
> /// variants we add to this enum.

Reply:
Not yet, it might only become a problem if we get in the realm of >50